pkg/connector/kairosdb.go

```diff
@@ -0,0 +1,494 @@
+// +build kairosdb
+
+package connector
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/facette/facette/pkg/catalog"
+	"github.com/facette/facette/pkg/config"
+	"github.com/facette/facette/pkg/logger"
+	"github.com/facette/facette/pkg/plot"
+	"github.com/facette/facette/pkg/utils"
+)
+
+const (
+	kairosdbURLVersion     string  = "/api/v1/version"
+	kairosdbURLMetricNames string  = "/api/v1/metricnames"
+	kairosdbURLMetricsTags string  = "/api/v1/datapoints/query/tags"
+	kairosdbURLQueryMetric string  = "/api/v1/datapoints/query"
+	kairosdbDefaultTimeout float64 = 10
+)
+
+type kairosdbSeriesEntry struct {
+	metric     string
+	tag        string
+	source     string
+	aggregator MetricsQueryAggregation
+}
+
+type MetricsQueryAggregation struct {
+	Name     string            `json:"name"`
+	Sampling map[string]string `json:"sampling"`
+}
+
+type MetricsQueryEntry struct {
+	Name        string                    `json:"name"`
+	Tags        map[string][]string       `json:"tags"`
+	Aggregators []MetricsQueryAggregation `json:"aggregators"`
+}
+
+type MetricsQueryResponse struct {
+	Sample_size int64                `json:"sample_size"`
```

Comment:
Am 18.01.2015 um 12:34 schrieb Vincent Batoufflet:
> In pkg/connector/kairosdb.go
> <https://github.com/facette/facette/pull/153#discussion-diff-23132659>:
> 
>> +type MetricsQueryResponse struct {
>> +	Sample_size int64                `json:"sample_size"`
> 
> Please use CamelCase notations for structures variables.

Yepp. Should be fixed in next commit.